server/src/main/java/com/objectcomputing/checkins/services/private_notes/PrivateNoteServicesImpl.java

```diff
@@ -0,0 +1,99 @@
+package com.objectcomputing.checkins.services.private_notes;
+
+import com.objectcomputing.checkins.services.checkins.CheckIn;
+import com.objectcomputing.checkins.services.checkins.CheckInServices;
+import com.objectcomputing.checkins.services.memberprofile.MemberProfile;
+import com.objectcomputing.checkins.services.memberprofile.MemberProfileServices;
+import com.objectcomputing.checkins.services.memberprofile.currentuser.CurrentUserServices;
+import com.objectcomputing.checkins.services.role.RoleType;
+import io.micronaut.security.utils.SecurityService;
+
+import javax.inject.Singleton;
+import javax.validation.constraints.NotNull;
+import java.util.UUID;
+
+@Singleton
+public class PrivateNoteServicesImpl implements PrivateNoteServices {
+
+    private final CheckInServices checkinServices;
+    private final PrivateNoteRepository privateNoteRepository;
+    private final MemberProfileServices memberProfileServices;
+    private final SecurityService securityService;
+    private final CurrentUserServices currentUserServices;
+
+    public PrivateNoteServicesImpl(CheckInServices checkinServices, PrivateNoteRepository privateNoteRepository,
+                                   MemberProfileServices memberProfileServices, SecurityService securityService,
+                                   CurrentUserServices currentUserServices) {
+        this.checkinServices = checkinServices;
+        this.privateNoteRepository = privateNoteRepository;
+        this.memberProfileServices = memberProfileServices;
+        this.securityService = securityService;
+        this.currentUserServices = currentUserServices;
+    }
+
+    @Override
+    public PrivateNote save(@NotNull PrivateNote privateNote) {
+        String workEmail = securityService != null ? securityService.getAuthentication().get().getAttributes().get("email").toString() : null;
+        MemberProfile currentUser = workEmail != null ? currentUserServices.findOrSaveUser(null, workEmail) : null;
```

Comment:
Can we use currentUserServices.getCurrentUser() instead of calling securityService?

Reply:
used currentUserServices.getCurrentUser()